From 16adf8611f9b6de41c7fa37f0222441720efe503 Mon Sep 17 00:00:00 2001
From: Mark Hatle <mhatle@xilinx.com>
Date: Mon, 6 Jun 2022 14:11:47 -0700
Subject: [PATCH] xrt_kernel.h: experimental/xrt_hw_context.h requries C++

Move the include under the cplusplus check to avoid the error:
   # error xrt_hwcontext is only implemented for C++

Signed-off-by: Mark Hatle <mhatle@xilinx.com>
---
 src/runtime_src/core/include/xrt/xrt_kernel.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/runtime_src/core/include/xrt/xrt_kernel.h b/src/runtime_src/core/include/xrt/xrt_kernel.h
index bbc9a30d5..93445dc81 100644
--- a/src/runtime_src/core/include/xrt/xrt_kernel.h
+++ b/src/runtime_src/core/include/xrt/xrt_kernel.h
@@ -11,9 +11,8 @@
 #include "xrt/xrt_device.h"
 #include "xrt/xrt_uuid.h"
 
-#include "experimental/xrt_hw_context.h"
-
 #ifdef __cplusplus
+# include "experimental/xrt_hw_context.h"
 # include "experimental/xrt_enqueue.h"
 # include <chrono>
 # include <cstdint>
-- 
2.17.1

